### PR TITLE
[WFCORE-2517] Coverity, Dereference after null check (Elytron subsystem)

### PR DESCRIPTION
**Pull request by ivassile**
wildfly-core: https://issues.jboss.org/browse/WFCORE-2517
JBEAP: https://issues.jboss.org/browse/JBEAP-9475

### CODE REVIEW

File: elytron/src/main/java/org/wildfly/extension/elytron/PolicyParser.java [hunk 1]
```diff
@@ -82,21 +82,26 @@ void readPolicy(ModelNode parentAddress, XMLExtendedStreamReader reader, List<Mo
...
-                    break;
-                case CUSTOM_POLICY:
-                    providerFound = defaultPolicy.equals(parseCustomPolicy(addPolicy, reader, operations)) || providerFound;
-                    break;
-                default:
-                    throw unexpectedElement(reader);
+        boolean providerFound = false;
+
+        if (defaultPolicy == null)
+        {
```
Comment (darranl) [inline elytron/src/main/java/org/wildfly/extension/elytron/PolicyParser.java]: Minor issue with the formatting.

File: elytron/src/main/java/org/wildfly/extension/elytron/PolicyParser.java [hunk 2]
```diff
@@ -82,21 +82,26 @@ void readPolicy(ModelNode parentAddress, XMLExtendedStreamReader reader, List<Mo
...
+            providerFound = true;
+        } else {
```
Comment (darranl) [inline elytron/src/main/java/org/wildfly/extension/elytron/PolicyParser.java]: This change doesn't feel quite correct - it skips the rest of the XML parsing are we sure that it should be skipped?

**Review by darranl — CHANGES_REQUESTED** [review]
Are we sure the XML parsing should be skipped?

**Comment by ivassile** [comment]
Submitted the requested changes.

**Review by kabir — APPROVED** [review]

**Pull request merged**